### DISCUSSION TITLE
Is it normal that changes don't transfer until file is saved?

### DISCUSSION THREAD

**Discussion by juandpinto** [overleaf-workshop/Overleaf-Workshop#106] [Q&A]
Basically the Q in the title. As I type or make changes in VSCode, those changes don't appear on Overleaf (though the collaborator cursor oddly does move). Changes only seem to transfer when I save the file in VSCode. Is this how the extension is designed to work, or am I experiencing a bug?

Also, a slightly different question: even when changes do transfer to Overleaf, my changes don't seem to be tracked, even if tracked changes is on for everyone. Is there a way to transfer my changes as tracked changes?

Awesome extension by the way!

**Comment by iamhyc** [comment]
> Basically the Q in the title. As I type or make changes in VSCode, those changes don't appear on Overleaf (though the collaborator cursor oddly does move). Changes only seem to transfer when I save the file in VSCode. Is this how the extension is designed to work, or am I experiencing a bug?

Yes, it is designed to upload the changes only when the file is saved, which is different from the Overleaf default behavior.

If you are in favor of the default behavior, you can configure the "auto save" option provided by vscode with "after delay".

![](https://github.com/iamhyc/Overleaf-Workshop/assets/9068301/39f90336-e93e-449c-a834-dd8fc49d3660)


> Also, a slightly different question: even when changes do transfer to Overleaf, my changes don't seem to be tracked, even if tracked changes is on for everyone. Is there a way to transfer my changes as tracked changes?
> 
> Awesome extension by the way!

The current release does no support track changes (tc). The feature is still under development here #95 .

However, as I could not figure out how the tc ID is generated (the function is closed source), creating a new tc is still impossible even the PR is merged.
